tools/wasp-cli/cli/wallet/providers/keychain.go

```diff
@@ -33,16 +37,38 @@ func LoadKeyChain(addressIndex uint32) wallets.Wallet {
 	return newInMemoryWallet(keyPair, addressIndex)
 }
 
-func CreateKeyChain() {
+func CreateKeyChain(overwrite bool) {
+	oldSeed, _ := config.GetKeyChain().GetSeed()
+
+	if len(oldSeed) == cryptolib.SeedSize && !overwrite {
+		log.Printf("You already have an existing seed inside your Keychain.\nCalling `init` will *replace* it with a new one.\n")
+		log.Printf("Run `wasp-cli init --overwrite` to continue with the initialization.\n")
+		log.Fatalf("The cli will now exit.")
+	}
+
 	seed := cryptolib.NewSeed()
 	err := config.GetKeyChain().SetSeed(seed)
 	log.Check(err)
 
-	log.Printf("New seed stored in the keychain.\n")
+	log.Printf("New seed stored inside the Keychain.\n")
 }
 
 func MigrateKeyChain(seed cryptolib.Seed) {
 	err := config.GetKeyChain().SetSeed(seed)
 	log.Check(err)
-	log.Printf("Seed migrated to keychain")
+	log.Printf("Seed migrated to Keychain.\nProceeding seed validation.\n")
+
+	kcSeed, err := config.GetKeyChain().GetSeed()
+	log.Check(err)
+
+	if reflect.DeepEqual(kcSeed[:], seed[:]) {
+		log.Printf("Seed has been successfully validated.\n")
+		config.RemoveSeedForMigration()
+		err = viper.WriteConfig()
+		log.Check(err)
+		log.Printf("Seed was removed from the wasp-cli.json\n")
```

Comment:
```suggestion
		log.Printf("Seed was removed from wasp-cli.json\n")
```

Also, maybe `"wasp-cli.json"` should be a constant, if it is not already.

Or, just say "the config file".